backend/app.ts

```diff
@@ -1,56 +1,110 @@
+
+app.use(passport.initialize());
 app.use(express.json());
 app.use(cors());
 app.use(express.urlencoded({ extended: false }));
 app.use(express.static(path.join(__dirname, "client/build")));
 
-passport.use(new LocalStrategy({ usernameField: "email" },
-  function (username, password, done) {
-    if (username !== "admin") {
-      return done(null, false, { message: "Username is wrong"});
-    }
-    if (password !== "password") {
-      return done(null, false, { message: "Password is wrong" });
-    }
-    return done(null, { username: "admin", message:"Login Successful" });
+// -------------------unprotected routes
+app.get("/unprotected", (request: Request, response: Response) => {
+  response.json({ msg: "unprotected" });
+});
+
+app.post("/signin", async (req, res, next) => {
+  try {
+    passport.authenticate(
+      "local",
+      (passportError: Error, user: Employee, info: any) => {
+        if (passportError || !user) {
+          res.status(400).json({ message: info.reason });
+          return;
+        }
+        req.login(user, { session: false }, (loginError) => {
+          if (loginError) {
+            res.send(loginError);
+            return;
+          }
+          const token = jwt.sign(
+            { id: user.id, email: user.email },
+            "jwt-secret-key",
+            { expiresIn: "1d" }
+          );
+          res.json({ token });
+        });
+      }
+    )(req, res);
+  } catch (error) {
+    console.error(error);
+    next(error);
+  }
+});
+
+app.post("/signup", async (req: Request, res: Response, next: NextFunction) => {
+  const insertClauseBuilder = (body: any): EmployeeInsert => {
+    const insertClause: EmployeeInsert = {
+      id: body.id,
+      email: body.email,
+      password: body.password,
+      firstName: body.firstName,
+      lastName: body.lastName,
+      position: body.position,
+      pronoun: body.pronoun,
+      createDate: body.date,
+      isManager:
+        // this part can only be boolean|undefined, or prisma will rasie type error.
+        body.isManager === "true"
+          ? true
+          : body.isManager === "false"
+          ? false
+          : undefined,
+    };
+
+    return insertClause;
+  };
+
+  try {
+    await createOneEmployee(insertClauseBuilder(req.body));
+    res.status(200).json({ message: `user ${req.body.email} is saved` });
+  } catch (e: any) {
+    next(new Error(e));
   }
-));
+});
+
+// check jwt. Router Above this line is unprotected, below this line is protected
+app.use(passport.authenticate("jwt", { session: false }));
 
-// -------------------routes
+// -------------------protected routes
 app.get("/", (request: Request, response: Response) => {
   response.json({ message: `Welcome to backend!!` });
 });
 
+//this home router returns the authenticated employee info
 app.get("/home", (request: Request, response: Response) => {
-  response.json({ message: `Welcome to the home page!!` });
+  response.json({ message: request.user });
 });
```

Comment:
this should be at endpoint `GET employee/profile` (should we have the obj be `{ user: request.user }` as well?)